### ISSUE TITLE
my anomaly map is weired

### ISSUE THREAD

**Issue by Lhy-opt** [guojiajeremy/Dinomaly#20]
Sorry to bother you again

I just trained a model on Mvtec-AD, and try to generate anomaly maps from test samples, but question here, the anomaly map is really weired

first please check my code:
`def test(item_list):  
    test_data_list = []
    
    device = 'cuda' if torch.cuda.is_available() else 'cpu'
    batch_size = 1
    image_size = 448
    crop_size = 392
    data_transform, gt_transform = get_data_transforms(image_size, crop_size)
    
    for i, item in enumerate(item_list):
        test_path = os.path.join(args.data_path, item)
        test_data = TeDataset(test_path, transform=data_transform, gt_transform=gt_transform, phase='test')
        test_data_list.append(test_data)
    
    test_data = ConcatDataset(test_data_list)
    test_dataloader = DataLoader(test_data, batch_size=batch_size, shuffle=False, num_workers=4)

    encoder_name = 'dinov2reg_vit_base_14'
    target_layers = [2, 3, 4, 5, 6, 7, 8, 9]
    fuse_layer_encoder = [[0, 1, 2, 3], [4, 5, 6, 7]]
    fuse_layer_decoder = [[0, 1, 2, 3], [4, 5, 6, 7]]
    encoder = vit_encoder.load(encoder_name)
    
    if 'small' in encoder_name:
        embed_dim, num_heads = 384, 6
    elif 'base' in encoder_name:
        embed_dim, num_heads = 768, 12
    elif 'large' in encoder_name:
        embed_dim, num_heads = 1024, 16
        target_layers = [4, 6, 8, 10, 12, 14, 16, 18]
    else:
        raise "Architecture not in small, base, large."

    bottleneck = []
    decoder = []

    bottleneck.append(bMlp(embed_dim, embed_dim * 4, embed_dim, drop=0))
    bottleneck = nn.ModuleList(bottleneck)

    for i in range(8):
        blk = VitBlock(dim=embed_dim, num_heads=num_heads, mlp_ratio=4.,
                       qkv_bias=True, norm_layer=partial(nn.LayerNorm, eps=1e-8),
                       attn=LinearAttention2)
        decoder.append(blk)
    decoder = nn.ModuleList(decoder)

    model = ViTill(encoder=encoder, bottleneck=bottleneck, decoder=decoder, target_layers=target_layers,
                    mask_neighbor_size=0, fuse_layer_encoder=fuse_layer_encoder, fuse_layer_decoder=fuse_layer_decoder)
    pretrained_weights = torch.load('./saved_results/te_torch1.2/model_10000.pth', map_location='cuda:0')

    model.load_state_dict(pretrained_weights)
    model = model.to(device)
    
    #test
    img_test = cv2.imread('000.png')
    img_test = cv2.resize(img_test, (392, 392))
    img_test = cv2.cvtColor(img_test, cv2.COLOR_BGR2RGB)
    img_test = torch.from_numpy(img_test).permute(2, 0, 1).unsqueeze(0).float() / 255.0
    img_test = img_test.to(device)
    with torch.no_grad():
        enc, dec = model(img_test)
        anomaly_map, _ = cal_anomaly_map(enc, dec, img_test.shape[-1], amap_mode='add')
        anomaly_map = np.expand_dims(anomaly_map, -1)
        
        min_val = np.min(anomaly_map)
        max_val = np.max(anomaly_map)
        anomaly_map = (anomaly_map - min_val) / (max_val - min_val)
        anomaly_map = cv2.applyColorMap(np.uint8(anomaly_map * 255), cv2.COLORMAP_JET)
        cv2.imwrite("anomaly_map.png", anomaly_map)

    return
`
all functions in the code come from your source code. But the input image and anomaly_map as bellow:

<img width="392" height="392" alt="Image" src="https://github.com/user-attachments/assets/8c4bf1e0-0c31-4a27-9653-e3139c2efeae" />

<img width="392" height="392" alt="Image" src="https://github.com/user-attachments/assets/977afa77-af39-4c5d-8dd9-cb845cf4964e" />

the inference based on your Mvtec-AD_pretrained model ,posted in issues, i don't know how to fix it, can you give me some answer?

**Comment by guojiajeremy** [comment]
It seems that you did not use image transform.

    data_transforms = transforms.Compose([
        transforms.Resize((size, size)),
        transforms.ToTensor(),
        transforms.CenterCrop(isize),
        transforms.Normalize(mean=mean_train,
                             std=std_train)])

Try to follow the original practice.
